rskj-core/src/main/java/co/rsk/net/eth/RskWireProtocol.java

```diff
@@ -151,29 +154,44 @@ public void channelRead0(final ChannelHandlerContext ctx, EthMessage msg) throws
      *************************/
 
     protected void processStatus(org.ethereum.net.eth.message.StatusMessage msg, ChannelHandlerContext ctx) throws InterruptedException {
-
         try {
-            Genesis genesis = GenesisLoader.loadGenesis(config, config.genesisInfo(), config.getBlockchainConfig().getCommonConstants().getInitialNonce(), true);
-            if (!Arrays.equals(msg.getGenesisHash(), genesis.getHash().getBytes())
-                    || msg.getProtocolVersion() != version.getCode()) {
+            byte protocolVersion = msg.getProtocolVersion();
+            byte versionCode = version.getCode();
+            if (protocolVersion != versionCode) {
                 loggerNet.info("Removing EthHandler for {} due to protocol incompatibility", ctx.channel().remoteAddress());
-                if (msg.getProtocolVersion() != version.getCode()){
-                    loggerNet.info("Protocol version {} - message protocol version {}", version.getCode(), msg.getProtocolVersion());
-                } else {
-                    loggerNet.info("Config genesis hash {} - message genesis hash {}", genesis.getHash(), msg.getGenesisHash());
-                }
+                loggerNet.info("Protocol version {} - message protocol version {}",
+                        versionCode,
+                        protocolVersion);
                 ethState = EthState.STATUS_FAILED;
                 recordEvent(EventType.INCOMPATIBLE_PROTOCOL);
                 disconnect(ReasonCode.INCOMPATIBLE_PROTOCOL);
                 ctx.pipeline().remove(this); // Peer is not compatible for the 'eth' sub-protocol
                 return;
             }
 
-            if (config.networkId() != msg.getNetworkId()) {
-                loggerNet.info("Different network received: config network ID {} - message network ID {}", config.networkId(), msg.getNetworkId());
+            int networkId = config.networkId();
+            int msgNetworkId = msg.getNetworkId();
+            if (msgNetworkId != networkId) {
+                loggerNet.info("Removing EthHandler for {} due to invalid network", ctx.channel().remoteAddress());
+                loggerNet.info("Different network received: config network ID {} - message network ID {}",
+                        networkId, msgNetworkId);
                 ethState = EthState.STATUS_FAILED;
                 recordEvent(EventType.INVALID_NETWORK);
                 disconnect(ReasonCode.NULL_IDENTITY);
+                ctx.pipeline().remove(this);
```

Comment:
Could you split this addition as a separate commit? In the future, if anything weird occurs b/c of this line we can track this change and at least know that it was a conscious change and not just a missing line between a bigger code reordering

Reply:
Ok!